pkg/util/aad/aad.go

```diff
@@ -6,29 +6,90 @@ package aad
 import (
 	"context"
 	"net/http"
+	"strings"
 	"time"
 
+	azgraphrbac "github.com/Azure/azure-sdk-for-go/services/graphrbac/1.6/graphrbac"
+	"github.com/Azure/go-autorest/autorest"
 	"github.com/Azure/go-autorest/autorest/adal"
 	"github.com/Azure/go-autorest/autorest/azure/auth"
 	"github.com/form3tech-oss/jwt-go"
 	"github.com/sirupsen/logrus"
 	"k8s.io/apimachinery/pkg/util/wait"
 
 	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/env"
+	"github.com/Azure/ARO-RP/pkg/util/azureclient/graphrbac"
 	"github.com/Azure/ARO-RP/pkg/util/refreshable"
 )
 
+type Manager interface {
+	GetToken(ctx context.Context) (*adal.ServicePrincipalToken, error)
+	GetServicePrincipalID(ctx context.Context) (string, error)
+}
+
+type manager struct {
+	log             *logrus.Entry
+	env             env.Interface
+	oc              *api.OpenShiftCluster
+	subscriptionDoc *api.SubscriptionDocument
+}
+
+func NewManager(log *logrus.Entry, env env.Interface, oc *api.OpenShiftCluster, subscriptionDoc *api.SubscriptionDocument) Manager {
```

Comment:
this should be written with a view to it being able to "just work" from the operator context - i.e. don't rely on oc, subscriptionDoc or maybe even env? (not sure).

Reply:
agreed